Enterprise-CMCS/macpro-mdct-mfp
# CMDCT 261 - Admin Review Functionality

### Description
https://jiraent.cms.gov/browse/CMDCT-261
This ticket completes the Admin Review functionality below:

1. Unlock functionality (Submitted reports only)

Enable Unlock and Approve Buttons
***  “Close” - Modal closes and user remains on the Review and Submit page
“Return to dashboard” takes the user to the [state specific admin dashboard](https://www.figma.com/file/oOJMUZerbafZefLQQ33GwI/MFP_Playground?type=design&node-id=3379%3A47897&mode=design&t=dF2v4U7WRZgbkAWH-1)

2. Approval Functionality

If the admin clicks the “Approve” button, a modal will appear that will require the admin type in “approve” (not case sensitive) and hit the “Approve” button inside the modal, which is validated on key stroke.
Update report to "Approved"
Redirect Admin to [state specific admin dashboard](https://www.figma.com/file/oOJMUZerbafZefLQQ33GwI/MFP_Playground?type=design&node-id=3379%3A47897&mode=design&t=dF2v4U7WRZgbkAWH-1)
Display updated status in [state specific admin dashboard](https://www.figma.com/file/oOJMUZerbafZefLQQ33GwI/MFP_Playground?type=design&node-id=3379%3A47897&mode=design&t=dF2v4U7WRZgbkAWH-1)


Unlock Report:
https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/14514294/aa88d1a0-e71d-45a3-818a-bba5f80dd93f

Approve Report:

https://github.com/Enterprise-CMCS/macpro-mdct-mfp/assets/14514294/351fe210-8403-499f-8e8d-6a094ec9681f


### Related ticket(s)


CMDCT- Subtask- https://jiraent.cms.gov/browse/CMDCT-261
Main Ticket - https://jiraent.cms.gov/browse/CMDCT-2995

---
### How to test
To test you will need to mock a submitted report.
1. In create.ts 
Within the Item object add or update the properties *status: "Submitted"* and "locked: true" 
2. Sign back in as a state user and make sure the report generated has a Submitted status.
3. Sign out and sign in as a admin user
4. Select the state the report is in
4.  View report and go to the "Review and Submit" page ( you should see the admin review section below)
5. Test out the "Unlock" functionality then refresh dev environment and repeat steps 2 and 3 before testing the "Approve" button. ( Once you unlock the report the buttons disable and your redirected to the dashboard)

Here's the criteria for how the buttons and modals should function:
https://jiraent.cms.gov/browse/CMDCT-261


### Important updates




---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
